src/Microsoft.Identity.Abstractions/DownstreamApi/IDownstreamApi.cs

```diff
@@ -236,5 +235,136 @@ public partial interface IDownstreamApi
             Action<DownstreamApiOptions>? downstreamApiOptionsOverride = null,
             CancellationToken cancellationToken = default)
             where TOutput : class;
+
+#if NET8_0_OR_GREATER
+        /// <summary>
+        /// Calls a downstream API consuming JSON with some data and returns data.
+        /// </summary>
+        /// <typeparam name="TInput">Input type.</typeparam>
+        /// <typeparam name="TOutput">Output type.</typeparam>
+        /// <param name="serviceName">Name of the service describing the downstream API. There can
+        /// be several configuration named sections mapped to a <see cref="DownstreamApiOptions"/>,
+        /// each for one downstream API. You can pass-in null, but in that case <paramref name="downstreamApiOptionsOverride"/>
+        /// needs to be set.</param>
+        /// <param name="input">Input parameter to the downstream web API.</param>
+        /// <param name="inputJsonTypeInfo">JSON serialization metadata for TInput</param>
+        /// <param name="outputJsonTypeInfo">JSON serialization metadata for TOutput</param>
+        /// <param name="downstreamApiOptionsOverride">Overrides the options proposed in the configuration described
+        /// by <paramref name="serviceName"/>.</param>
+        /// <param name="user">[Optional] Claims representing a user. This is useful in platforms like Blazor
+        /// or Azure Signal R, where the HttpContext is not available. In other platforms, the library
+        /// will find the user from the HttpContext.</param>
+        /// <param name="cancellationToken">Cancellation token.</param>
+        /// <returns>The value returned by the downstream web API.</returns>
+        /// <example>
+        /// A list method that returns an IEnumerable&lt;MyItem&gt;&gt;.
+        /// <code>
+        /// public Task&lt;IEnumerable&lt;MyItem&gt;&gt; GetAsync()
+        /// {
+        ///  return _downstreamWebApi.CallWebApiForUserAsync&lt;object, IEnumerable&lt;MyItem&gt;&gt;(
+        ///         ServiceName,
+        ///         null,
+        ///         options =>
+        ///         {
+        ///           options.RelativePath = $"api/todolist";
+        ///         });
+        /// }
+        /// </code>
+        ///
+        /// Example of editing.
+        /// <code>
+        /// public Task&lt;MyItem&gt; EditAsync(MyItem myItem)
+        /// {
+        ///   return _downstreamWebApi.CallWebApiForUserAsync&lt;MyItem, MyItem&gt;(
+        ///         ServiceName,
+        ///         nyItem,
+        ///         options =>
```

Comment:
You might want to update this code snippet (which goes in the reference documentation). See the Examples paragraph of https://learn.microsoft.com/en-us/dotnet/api/microsoft.identity.abstractions.idownstreamapi.callapiforuserasync?view=msal-model-dotnet-latest for instance